Submitted & accepted upstream for a release after 12.2:
https://gitlab.freedesktop.org/pulseaudio/pulseaudio/merge_requests/111/commits

From 4be2625ef8cc3270827e522efadc1fb78439410d Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Wed, 26 Jun 2019 18:32:40 -0700
Subject: [PATCH] core-util: Use /proc/fd on Solaris as well in pa_close_all

Gets rid of > 65,000 unnecessary close() syscalls

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 src/pulsecore/core-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index b37f7f8cb..f2748aeb1 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -2553,7 +2553,7 @@ int pa_close_allv(const int except_fds[]) {
     struct rlimit rl;
     int maxfd, fd;
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__sun)
     int saved_errno;
     DIR *d;
 
-- 
2.19.2

